tasks/libs/common/junit_upload_core.py

```diff
@@ -372,14 +365,9 @@ def repack_macos_junit_tar(workflow_conclusion, infile, outfile):
         infp.extractall(tempd)
 
         # write the proper job url and job name
-        with open(os.path.join(tempd, JOB_URL_FILE_NAME), "w") as fp:
-            fp.write(os.environ.get("CI_JOB_URL", ""))
-        with open(os.path.join(tempd, TAGS_FILE_NAME)) as fp:
-            tags = fp.read()
-        job_name = os.environ.get("CI_JOB_NAME", "")
-        tags = tags.replace("ci.job.name:", f"ci.job.name:{job_name}")
-        with open(os.path.join(tempd, TAGS_FILE_NAME), "w") as fp:
-            fp.write(tags)
+        with open(os.path.join(tempd, JOB_ENV_FILE_NAME), "w") as fp:
+            fp.write(f'CI_JOB_URL={os.environ.get("CI_JOB_URL", "")}')
+            fp.write(f'CI_JOB_NAME={os.environ.get("CI_JOB_NAME", "")}')
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏You could use `writelines`